src/client.cpp

```diff
@@ -533,9 +547,139 @@ PVR_ERROR GetChannelGroupMembers(ADDON_HANDLE handle, const PVR_CHANNEL_GROUP &g
+  return PVR_ERROR_NO_ERROR;
+}
+
+/* live stream functions */
+bool OpenLiveStream(const PVR_CHANNEL &channel)
+{
+  if (!VuData || !VuData->IsConnected())
+    return false;
+
+  if (!VuData->OpenLiveStream(channel))
+    return false;
+
+  //const Settings &settings = DvbData->GetSettings();
+
+  /* queue a warning if the timeshift buffer path does not exist */
+  if (g_iEnableTimeshift != TIMESHIFT_OFF
+      && !XBMC->DirectoryExists(g_strTimeshiftBufferPath.c_str()))
+    XBMC->QueueNotification(QUEUE_ERROR, LocalizedString(30514).c_str());
+
+  std::string streamURL = VuData->GetLiveStreamURL(channel);
+  strReader = new StreamReader(streamURL, g_iReadTimeout);
+  if (g_iEnableTimeshift == TIMESHIFT_ON_PLAYBACK)// || g_iEnableTimeshift == TIMESHIFT_ON_PAUSE)
+    strReader = new TimeshiftBuffer(strReader, g_strTimeshiftBufferPath, g_iReadTimeout);
+  
+  return strReader->Start();
+}
+
+void CloseLiveStream(void)
+{
+  VuData->CloseLiveStream();
+  SAFE_DELETE(strReader);
+}
+
+bool IsRealTimeStream()
+{
+  return (strReader) ? strReader->IsRealTime() : false;
+}
+
+bool CanPauseStream(void)
+{
+  if (!VuData)
+    return false;
+
+  //const Settings &settings = DvbData->GetSettings();
+  if (g_iEnableTimeshift != TIMESHIFT_OFF && strReader)
+    return (strReader->IsTimeshifting() || XBMC->DirectoryExists(g_strTimeshiftBufferPath.c_str()));
+
+  return false;
+}
+
+bool CanSeekStream(void)
+{
+  if (!VuData)
+    return false;
+
+  // pause button seems to check CanSeekStream() too
+  //return (strReader && strReader->IsTimeshifting());
+
+  return (g_iEnableTimeshift != TIMESHIFT_OFF);
+}
+
+int ReadLiveStream(unsigned char *buffer, unsigned int size)
+{
+  return (strReader) ? strReader->ReadData(buffer, size) : 0;
+}
+
+long long SeekLiveStream(long long position, int whence)
+{
+  return (strReader) ? strReader->Seek(position, whence) : -1;
+}
+
+long long LengthLiveStream(void)
+{
+  return (strReader) ? strReader->Length() : -1;
+}
+
+bool IsTimeshifting(void)
+{
+  return (strReader && strReader->IsTimeshifting());
+}
+
+PVR_ERROR GetStreamTimes(PVR_STREAM_TIMES *times)
+{
+    
+
+  if (!times)
+    return PVR_ERROR_INVALID_PARAMETERS;
+  if (strReader)
+  {
+    times->startTime = strReader->TimeStart();
+    times->ptsStart  = 0;
+    times->ptsBegin  = 0;
+    times->ptsEnd    = (!strReader->IsTimeshifting()) ? 0
+      : (strReader->TimeEnd() - strReader->TimeStart()) * DVD_TIME_BASE;
+    
+    return PVR_ERROR_NO_ERROR;
+  }
+  return PVR_ERROR_NOT_IMPLEMENTED;
+}
+
+void PauseStream(bool paused)
+{
+  if (!VuData)
+    return;
+
+  /* start timeshift on pause */
+  //const Settings &settings = DvbData->GetSettings();
+  if (paused && g_iEnableTimeshift == TIMESHIFT_ON_PAUSE
+      && strReader && !strReader->IsTimeshifting()
+      && XBMC->DirectoryExists(g_strTimeshiftBufferPath.c_str()))
+  {
+    strReader = new TimeshiftBuffer(strReader, g_strTimeshiftBufferPath ,g_iReadTimeout);
+    (void)strReader->Start();
```

Comment:
what's this leading  `(void)` good for?

Reply:
Casting a function to void is a hint that the return value is not used and that the programmer is aware of it. It might also suppress a warning if lint is used.
TimeshiftBuffer->Start() returns a bool, which is ignored here.